test/Storage.test.js

```diff
@@ -111,6 +112,34 @@ describe("Storage", function () {
       const expectedBalance = (collateralAmount * (100 - slashPercentage)) / 100
       expect(await storage.balanceOf(host.address)).to.equal(expectedBalance)
     })
+
+    it("fails to mark proof as missing when cancelled", async function () {
+      await storage.fillSlot(slot.request, slot.index, proof)
+      await advanceTimeTo(request.expiry + 1)
+      let missedPeriod = periodOf(await currentTime())
+      await expect(
+        storage.markProofAsMissing(slotId(slot), missedPeriod)
+      ).to.be.revertedWith("Request was cancelled")
+    })
+  })
+  describe("contract state", function () {
```

Comment:
I wonder whether we shouldn't also change `isProofRequired()`, `willProofBeRequired()`, and `proofEnd()` when request is cancelled.

Reply:
I believe this is covered in https://github.com/status-im/dagger-contracts/pull/19.

Reply:
`proofEnd` is not covered by state checks for cancelled or finished. `proofEnd` returns a `uint256`, so what should the value be if a contract is cancelled?

Reply:
I would expect `proofEnd` to change to the current time when the state becomes cancelled or finished. This has the added benefit that `removeEndedContracts()` in `proving.nim` would nicely clean up cancelled or finished contracts.